callback/src/main/java/com/iluwatar/callback/Task.java

```diff
@@ -24,15 +24,20 @@
  */
 package com.iluwatar.callback;
 
-import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
 
 /** Template-method class for callback hook execution. */
 public abstract class Task {
 
-  /** Execute with callback. */
+  /** Execute the task and asynchronously call the callback method upon completion. */
   final void executeWith(Callback callback) {
-    execute();
-    Optional.ofNullable(callback).ifPresent(Callback::call);
+    CompletableFuture.runAsync(
+        () -> {
+          execute();
+          if (callback != null) {
+            callback.call();
+          }
+        });
```

Comment:
In general, looks good! However, I'd like to keep the synchronous callbacks (previous functionality) as well.

Could we demonstrate both sychronous and asynchronous callback in the example? The guidance would be something like

- If your operation is short and straightforward, keep it synchronous.
- If the operation can block or is resource-intensive, prefer asynchronous callbacks to maintain application responsiveness.